chain/client/src/sync.rs

```diff
@@ -744,20 +750,25 @@ impl StateSync {
                         update_sync_status = true;
                         *shard_sync_download = ShardSyncDownload {
                             downloads: vec![],
-                            status: ShardSyncStatus::StateDownloadFinalize,
+                            status: ShardSyncStatus::StateDownloadDownloaded,
                         };
                     }
                 }
-                ShardSyncStatus::StateDownloadFinalize => {
+                ShardSyncStatus::StateDownloadDownloaded => {
```

Comment:
Maybe this stage can be called `StateDownloadApplyParts`? I think the names of all other stages describe what it will do in this stage, not what has been done. Just to keep the naming consistent

Reply:
But we are in this state only for very short time - when we've downloaded parts but not yet scheduled them to be applied

Reply:
I see what you mean. Sorry for being nit picky, but maybe the following way is clearer. We can move the waiting logic to this stage 

```
let result = self.state_parts_apply_results.remove(&shard_id);
                    if let Some(result) = result {
```
and only proceed to the next stage once the result is ready. This way, it is clear that at one stage, we are applying the state parts, the next one we are doing the post processing after state parts being applied

Reply:
I am not sure what you mean. Waiting logic is already in the stage code of which you cited

Reply:
Maybe we can rename this one to StateDownloadScheduling

Reply:
And I am not consider this not picking, I am all for better naming